Change reference url

tqwewe opened the pull request:
Added two new functions.

- `SetRef(path string)`
- `SetUR(url string)`

**SetRef(path string)**
Edits the url for a firebase reference and changes the current path to a new path.

**SetURL(url string)**
Changes the complete URL of an existing firebase reference.

These functions can be useful in a number of ways with one being the fact that you no longer need to re-auth each time you change a firebase reference.

tqwewe committed:
Added SetRef() and SetURL() funcs

tqwewe committed:
Added check for '/' as prefix

tqwewe commented:
Example of use:
```
f := firego.New("https://my-firebase-app.firebaseIO.com", nil)
f.Auth("xxxxxxxxxxxxxxxxxxxxx")

f.SetRef("users/204493/age")

var age string
f.Value(&age)
```

zabawaba99 commented on firebase.go:
```diff
@@ -105,6 +105,23 @@ func (fb *Firebase) Unauth() {
+func (fb *Firebase) SetRef(path string) {
+	parsedURL, err := _url.Parse(fb.url)
+	if err != nil {
+		panic(err)
```
Instead of panicking can you return an error here and let the user handle it?

zabawaba99 commented on firebase.go:
```diff
@@ -105,6 +105,23 @@ func (fb *Firebase) Unauth() {
+	if err != nil {
+		panic(err)
+	}
+	if strings.Index(path, "/") == 0 {
```
I would suggest doing something like

```go
path = strings.Trim(path, "/")
```

So that we can trim the leading and trailing slashes

zabawaba99 reviewed (commented):
Would you mind adding a test for this function?

tqwewe committed:
Updated SetRef() Func

Improved the SetRef() function and removed the SetURL() function (already added in change-reference branch).

zabawaba99 commented:
Closed on favor of #69
